curvefs/src/metaserver/s3compact_wq_impl.cpp

```diff
@@ -351,41 +431,39 @@ int S3CompactWorkQueueImpl::WriteFullChunk(
     return 0;
 }
 
-void S3CompactWorkQueueImpl::CompactChunks(
-    std::shared_ptr<InodeStorage> inodeStorage, InodeKey inodeKey,
-    PartitionInfo pinfo, std::shared_ptr<CopysetNodeWrapper> copysetNodeWrapper,
-    std::shared_ptr<S3AdapterManager> s3adapterManager,
-    std::shared_ptr<S3InfoCache> s3infoCache) {
-    auto cleanup = absl::MakeCleanup([this, inodeKey]() {
+void S3CompactWorkQueueImpl::CompactChunks(const struct S3CompactTask& task) {
```

Comment:
This function is too too long, please split it into small functions.

Reply:
done